delivery/app.py

```diff
@@ -6,6 +6,44 @@
 from delivery.error_handling import CustomError, WrongIdError
 from flasgger import Swagger
 from marshmallow import Schema, fields, ValidationError
+from abc import ABC, abstractmethod
+
+
+class AbstractRestaurantRepository(ABC):
+    @abstractmethod
+    def save(self, content, restaurant_id):
+        pass
+
+    @abstractmethod
+    def get_all(self):
+        pass
+
+    @abstractmethod
+    def get_by_id(self, id):
+        pass
+
+
+class MemoryRestaurantRepository(AbstractRestaurantRepository):
+    def __init__(self, path):
+        try:
+            with open(path, "r") as read_file:
+                data = json.load(read_file)
+                self.restaurants = data['restaurants']
+        except TypeError:
+            raise TypeError("Set environment variable: PATH_FOR_INITIAL_DATA")
+
+    def save(self, content, restaurant_id=None):
```

Comment:
it might be better to have 2 methods: `create` and `update`

Reply:
fixed